src/tail/tail.rs

```diff
@@ -187,7 +187,7 @@ pub fn uumain(args: Vec<String>) -> i32 {
 
     let files = given_options.free;
 
-    if files.is_empty() {
+    if uucore::coreopts::is_stdin(&files) {
```

Comment:
Apparently GitHub didn't record my other comment, so I'm going to write a shorter version.

Basically, for each file, you need to check if it is `-` because they are processed in the order they are given.  This means that if you do something like `tail file - - file < file`, it should process `file`, then handle `stdin`, then `stdin` again (but since it's already been processed there shouldn't be any data), then `file` again.